tests/integration/nm/route_test.py

```diff
@@ -0,0 +1,168 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+import copy
+
+from libnmstate import metadata
+from libnmstate.nm import ipv4 as nm_ipv4
+from libnmstate.nm import ipv6 as nm_ipv6
+from libnmstate.nm import connection as nm_connection
+from libnmstate.schema import Route
+
+IPV4_ROUTE1 = {
+    Route.DESTINATION: '198.51.100.0/24',
+    Route.METRIC: 103,
+    Route.NEXT_HOP_ADDRESS: '192.0.2.1',
+    Route.TABLE_ID: 50
+}
+
+IPV4_ROUTE2 = {
+    Route.DESTINATION: '203.0.113.0/24',
+    Route.METRIC: 103,
+    Route.NEXT_HOP_ADDRESS: '192.0.2.2',
+    Route.TABLE_ID: 51
+}
+
+IPV6_ROUTE1 = {
+    Route.DESTINATION: '2001:db8:a::/64',
+    Route.METRIC: 103,
+    Route.NEXT_HOP_ADDRESS: '2001:db8:1::a',
+    Route.TABLE_ID: 50
+}
+
+IPV6_ROUTE2 = {
+    Route.DESTINATION: '2001:db8:b::/64',
+    Route.METRIC: 103,
+    Route.NEXT_HOP_ADDRESS: '2001:db8:1::b',
+    Route.TABLE_ID: 51
+}
+
+
+def test_ipv4_route_add_route():
+    setting_ip = nm_ipv4.create_setting({
+        'enabled': True,
+        metadata.ROUTES: [IPV4_ROUTE1, IPV4_ROUTE2]
+    }, base_con_profile=None)
+    assert (list(_nm_route_to_dict(r) for r in setting_ip.props.routes) ==
+            [IPV4_ROUTE1, IPV4_ROUTE2])
+
+
+def test_ipv6_route_add_route():
+    setting_ip = nm_ipv6.create_setting({
+        'enabled': True,
+        metadata.ROUTES: [IPV6_ROUTE1, IPV6_ROUTE2]
+    }, base_con_profile=None)
+    assert (list(_nm_route_to_dict(r) for r in setting_ip.props.routes) ==
+            [IPV6_ROUTE1, IPV6_ROUTE2])
+
+
+def test_ipv4_clear_route():
+    setting_ip = nm_ipv4.create_setting({
+        'enabled': True,
+        metadata.ROUTES: [IPV4_ROUTE1, IPV4_ROUTE2]
+    }, base_con_profile=None)
+    con_profile = nm_connection.ConnectionProfile()
+    con_profile.create([setting_ip])
+    new_setting_ip = nm_ipv4.create_setting({
+        'enabled': True,
+        metadata.ROUTES: []
+    }, base_con_profile=con_profile.profile)
+    assert not list(_nm_route_to_dict(r) for r in new_setting_ip.props.routes)
+
+
+def test_ipv6_clear_route():
+    setting_ip = nm_ipv6.create_setting({
+        'enabled': True,
+        metadata.ROUTES: [IPV6_ROUTE1, IPV6_ROUTE2]
+    }, base_con_profile=None)
+    con_profile = nm_connection.ConnectionProfile()
+    con_profile.create([setting_ip])
+    new_setting_ip = nm_ipv6.create_setting({
+        'enabled': True,
+        metadata.ROUTES: []
+    }, base_con_profile=con_profile.profile)
+    assert not list(_nm_route_to_dict(r) for r in new_setting_ip.props.routes)
+
+
+def test_ipv4_route_add_route_without_metric():
+    route_with_default_metric = copy.deepcopy(IPV4_ROUTE1)
```

Comment:
I believe the code is pretty understandable. You set a `route_without_metric`, then you should get
a `route_with_default_metric`.
The `expected_route` seems confusing to me.
Does route send to `create_setting()` expected, or the route with metric is expected?

Reply:
You have desired, expected and actual.
It took me too long to understand who is what.

It is clear to you because you wrote it, it is less clear to me as a reviewer of your code and therefore I expressed it in the feedback by proposing to change it a little bit.